sw/device/tests/dif/dif_otbn_smoketest.c

```diff
@@ -298,37 +110,24 @@ static void test_barrett384(void) {
  * The entry point wrap_err_test() is called, no arguments are passed or results
  * returned.
  */
-static void test_err_test(void) {
-  load_app(&kOtbnAppErrTest);
-
-  int t_start = ibex_mcycle_read();
-
-  LOG_INFO("Calling wrap_err_test()");
-  call_function(&kOtbnFuncErrTestWrapErrTest);
-
-  otbn_wait_for_done();
+static void test_err_test(otbn_t *otbn_ctx) {
+  CHECK(otbn_load_app(otbn_ctx, kOtbnAppErrTest) == kOtbnOk);
 
-  int t_end = ibex_mcycle_read();
-  LOG_INFO("Function execution on OTBN took %d cycles (end-to-end).",
-           t_end - t_start);
-
-  check_otbn_err_code(kDifOtbnErrCodeBadDataAddr);
+  CHECK(otbn_call_function(otbn_ctx, kOtbnFuncErrTestWrapErrTest) == kOtbnOk);
+  CHECK(otbn_busy_wait_for_done(otbn_ctx) == kOtbnExecutionFailed);
```

Comment:
Can we keep the check the expected error code is seen (expected error bit is set once my error bit PR is merged)?

In particular for the error bits there's a few points where things can get mixed up, we do have checking in various places for this but this is the only end to end test that can show a particular error in OTBN execution ends up setting the correct error bit as visible from Ibex.

Reply:
I'd expect in most cases the error code to be irrelevant to software, hence the optimization for this case. I see two solutions:

* Explicitly query the error code using dif_otbn_get_err_code(). That's already possible today and can be used in the test if you'd like to. That'll cause a second read of the ERR_CODE register (right now, we could optimize this behind the scenes).
* 
* Provide an out argument to otbn_call_function() to (optionally) return the error code.

I'd go with the first option for now until we have more software written and a better understanding of what we actually need for that.

Reply:
> I'd go with the first option for now

Works for me